### ISSUE TITLE
Installation failed: this should not have happened

### ISSUE THREAD

**Issue by edricantoine** [sukeesh/Jarvis#926]
When I cloned the repository and ran the command python installer from the terminal, I recieve this error message:

Installation failed with unexpected error - This should not have happened.
Please check logs at "/var/folders/yl/1y_2z49d7ks9xn5dscy_t9800000gn/T/tmpcu5fq7oc". If you open a bug report, please include this file.

I have tried both python installer and python3 installer, but get the same message each time.
I am on Mac OS, if that helps any. 
Any support would be appreciated. Thanks!

**Comment by pnhofmann** [comment]
Can you upload this file:

Please check logs at "/var/folders/yl/1y_2z49d7ks9xn5dscy_t9800000gn/T/tmpcu5fq7oc". If you open a bug report, please include this file.

**Comment by edricantoine** [comment]
I am not sure where to find this file, as I can't seem to find the yl folder in the project directory.

**Comment by edricantoine** [comment]
Also, when I try to run source setup.h, I get a much lengthier error message, but I think this is the most important part of it:

ERROR: Can not execute `setup.py` since setuptools is not available in the build environment.\n      [end of output]\n  \n  note: This error originates from a subprocess, and is likely not a problem with pip.\nerror: metadata-generation-failed\n\n\xd7 Encountered error while generating package metadata.\n\u2570\u2500> See above for output.\n\nnote: This is an issue with the package mentioned above, not pip.\nhint: See above for details.\n', 1891, 1892, 'ordinal not in range(128)')

**Comment by edricantoine** [comment]
Could someone get back to me on this issue? I would like to contribute to this project but need to get the application running on my end first.
Thank you!

**Comment by pnhofmann** [comment]
Sorry, it's a bit complicated for me since I don't have a Mac. But Okay; it says it is missing setuptools for some reason. Can you try:

```
cd JARVIS_FOLDER
source env/bin/activate
python3 -m ensurepip --default-pip
python3 -m pip install --upgrade pip setuptools wheel
# and if above command works:
pip3 install -U -r installer/requirements.txt
```

**Comment by edricantoine** [comment]
Hello,
I followed your given instructions. Once I entered the virtual environment, the ensurepip and pip install commands both worked, and I double-checked to make sure that setuptools was installed, but the last command still did not work.
This is the error message I recieved:
<img width="567" alt="Screen Shot 2022-02-28 at 9 39 55 AM" src="https://user-images.githubusercontent.com/72291309/156031665-f4fa83fd-a255-4573-b947-e82e142ffca4.png">

Thank you for your help!

**Comment by AngelCode-dev** [comment]
I'm having (probably) the same error, I don't know what else to attach other than the file it said to open and a picture of the error. 
![image](https://user-images.githubusercontent.com/85044270/156692812-688feb69-970b-4127-a315-18f0c670027c.png) 
[tmpk2cb0gla.txt](https://github.com/sukeesh/Jarvis/files/8182831/tmpk2cb0gla.txt)

**Comment by pnhofmann** [comment]
Okay, funny - the error is different, but the problem appears to be in pathlib, too.

https://pypi.org/project/pathlib/

I just noticed two things:

* In Python 3.4, pathlib is now part of the standard library. 
* Attention: this backport module isn’t maintained anymore. If you want to report issues or contribute patches, please consider the [pathlib2](https://pypi.python.org/pypi/pathlib2/) project instead.

So basically:

Edit installer/requirements.txt

Replace line

pathlib

with 

pathlib2; python_version < '3.4'

And check if it works ;).

**Comment by edricantoine** [comment]
Worked perfectly! Thanks for the fix!

**Comment by edricantoine** [comment]
I am unfortunately running into another issue now. Running python3 installer works now, but when I try to run source setup.sh as advised in the readme for mac users, I am getting this error:
<img width="569" alt="image" src="https://user-images.githubusercontent.com/72291309/156813116-71a781e9-45b7-4bf5-80e4-4e8009bc7318.png">

Seems to be a simple syntax error, though... I'll see if I can possibly try to fix it.

**Comment by shadow7412** [comment]
This worked for me as well. I've noted it's fixed on the master branch - but it hasn't been on the dev branch yet...

**Comment by JonB5** [comment]
Hey guys, I have tried all of the above solutions and even checked the requirements.txt file and am still having a similar  issue as the people above except my output message says:

"Installation failed with unexpected error - This should not have happened.
Please check logs at "/var/folders/nl/2m7v3tf9797csq4ddcv2rdrc0000gn/T/tmp6povwn_2". If you open a bug report, please include this file."
